File: Data Structures & Algorithms/concatenation-of-array/submission-1.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int nums_length = nums.length;
        int[] concatenated_nums = new int[2*nums_length];
        for (int i=0; i<nums_length; i++){
            concatenated_nums[i] = nums[i];
            concatenated_nums[i+nums_length] = nums[i];
        }
        return concatenated_nums;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int pivot;
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int m = l + (r - l)/2;
            if (nums[m] > nums[r]) l = m + 1;
            else r = m;
        }

        pivot = l;
        l = 0;
        r = nums.size() - 1;

        if (target >= nums[pivot] && target <= nums[r]) l = pivot;
        else r = pivot - 1;

        while (l <= r) {
            int m = l + (r-l)/2;
            if (target == nums[m]) return m;
            else if (target > nums[m]) l = m + 1;
            else r = m - 1;
        }
        
        return -1;
    }
};
